File: patch_server.py
import re

with open('server.ts', 'r') as f:
    content = f.read()

import_statement = "import investigationRoutes from './src/routes/investigation.js';\n"
content = content.replace("import evidenceRoutes from './src/routes/evidence.js';", "import evidenceRoutes from './src/routes/evidence.js';\n" + import_statement)

route_statement = "app.use('/api/investigation', investigationRoutes);\n"
content = content.replace("app.use('/api/evidence', evidenceRoutes);", "app.use('/api/evidence', evidenceRoutes);\n" + route_statement)

with open('server.ts', 'w') as f:
    f.write(content)

File: patch-server.mjs
import fs from 'fs';
let content = fs.readFileSync('server.ts', 'utf8');

if (!content.includes('/api/workspaces')) {
  content = content.replace(
    "import searchRoutes from \"./src/routes/search.js\";",
    "import searchRoutes from \"./src/routes/search.js\";\nimport workspacesRoutes from \"./src/routes/workspaces.js\";"
  );
  content = content.replace(
    "app.use(\"/api/search\", searchRoutes);",
    "app.use(\"/api/search\", searchRoutes);\napp.use('/api/workspaces', workspacesRoutes);"
  );
  fs.writeFileSync('server.ts', content);
}
